src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -26,7 +26,7 @@
 
   <PropertyGroup>
     <GenerateAssemblyInfo Condition="'$(GenerateAssemblyInfo)'==''">true</GenerateAssemblyInfo>
-    <BuiltByString Condition="'$(BuiltByString)'==''">%20built by: $(COMPUTERNAME)-$(USERNAME)</BuiltByString>
+    <BuiltByString Condition="'$(BuiltByString)'=='' AND '$(OSEnvironment)'=='Windows_NT'">%20built by: $(COMPUTERNAME)-$(USERNAME)</BuiltByString>
```

Comment:
For simplicity I think I would just move this to inside the target as well, and just always default to those properties for all OS's and for then for non-windows call those tools only if those properties aren't set.